src/field_specials.c

```diff
@@ -984,7 +985,7 @@ void FieldShowRegionMap(void)
 
 static bool32 IsBuildingPCTile(u32 tileId)
 {
-    return gMapHeader.mapLayout->primaryTileset == &gTileset_Building && (tileId == METATILE_Building_PC_On || tileId == METATILE_Building_PC_Off);
+    return (UNPACK_BEHAVIOR(GetMetatileAttributesById(tileId)) == MB_PC);
```

Comment:
I'm not 100% sure, but I think `MetatileBehavior_IsPC` might do this?
E: Ah, not quite. It just does the `== MB_PC` part. But I suspect there's a more idiomatic function to get the behavior?